cabal-install/src/Distribution/Client/InstallPlan.hs

```diff
@@ -206,12 +214,12 @@ instance
   => IsNode (GenericPlanPackage ipkg srcpkg)
   where
   type Key (GenericPlanPackage ipkg srcpkg) = UnitId
-  nodeKey (PreExisting ipkg) = nodeKey ipkg
-  nodeKey (Configured spkg) = nodeKey spkg
-  nodeKey (Installed spkg) = nodeKey spkg
-  nodeNeighbors (PreExisting ipkg) = nodeNeighbors ipkg
-  nodeNeighbors (Configured spkg) = nodeNeighbors spkg
-  nodeNeighbors (Installed spkg) = nodeNeighbors spkg
+  nodeKey (PreExisting ipkg _) = nodeKey ipkg
+  nodeKey (Configured spkg _) = nodeKey spkg
+  nodeKey (Installed spkg _) = nodeKey spkg
```

Comment:
And here we lose this information as well..., a UnitId refers only to a package, not the Scope in which we use this UnitId